server/src/modules/post/post.service.ts

```diff
@@ -117,12 +144,16 @@ export class PostService {
     })
 
     if (!posts) {
-      return []
+      return { posts: [], totalItems: 0 }
     } else if (withAnswers) {
-      return returnPosts
+      return this.getPaginatedPosts(returnPosts, page, size)
     } else {
       // posts without answers
-      return this.filterPostsWithoutAnswers(posts)
+      return this.getPaginatedPosts(
+        await this.filterPostsWithoutAnswers(posts),
```

Comment:
nit - just as good practice and to make it clearer that we await on this before moving to `this.getPaginatedPosts()`, I would have this on its own line.